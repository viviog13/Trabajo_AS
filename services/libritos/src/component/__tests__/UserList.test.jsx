import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
 
import UsersList from '../UsersList';
 
const users = [
  {
	'active': true,
	'autor': 'Susi',
	'id': 1,
	'titulo': 'Mil caeran',
	'añodepublicacion': '1990',
	'editorial': 'Ni idea',
	'generoliterario': 'Accion'
  },
  {
	'active': true,
	'autor': 'Mario',
	'id': 2,
	'titulo': 'La casa verde',
	'añodepublicacion': '1966',
	'editorial': 'Six Barral',
	'generoliterario': 'Ficcion'
  }
];

test('UsersList renders properly', () => {
  const wrapper = shallow(<UsersList users={users}/>);
  const element = wrapper.find('h4');
  expect(element.length).toBe(2);
  expect(element.get(0).props.children).toBe('Mil caeran');
});

test('UsersList renders a snapshot properly', () => {
  const tree = renderer.create(<UsersList users={users}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
