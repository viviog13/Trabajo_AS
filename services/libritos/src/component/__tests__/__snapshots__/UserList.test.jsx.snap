// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders a snapshot properly 1`] = `
<div>
  <h4
    className="box title is-4"
  >
    Mil caeran
  </h4>
  <h4
    className="box title is-4"
  >
    La casa verde
  </h4>
</div>
`;
